src/vs/platform/product/common/product.ts

```diff
@@ -47,7 +41,7 @@ else if (typeof require?.__$__nodeRequire === 'function') {
 	// want to have it running out of sources so we
 	// read it from package.json only when we need it.
 	if (!product.version) {
-		const pkg = require.__$__nodeRequire(joinPath(rootPath, 'package.json').fsPath) as { version: string };
+		const pkg = globalThis._VSCODE_PACKAGE_JSON as { version: string };
```

Comment:
This is lazy to avoid a `require(package.json)` when running because a built time we add `version` to the `product.json`. The only exception is running out of sources for why this is called here.

Reply:
In the ESM future there will be no more way to synchronously load anything. So, the only way forward will be something like `_VSCODE_PRODUCT` where the bootstrapper or embedder does these conditional loads. I suggest that we merge this with the `vscode.context` merging effort